tests/tests_accept_eula.yml

```diff
@@ -13,9 +13,13 @@
           include_role:
             name: linux-system-roles.mssql
 
+        - name: Unreachable task
+          fail:
+            msg: The above task must fail
+
       rescue:
         - name: Assert that the role failed with variables undefined
           assert:
             that: >-
-              "You must define the following variables to set up MSSQL" in
-              __mssql_variables_evaluation.msg[0]
+              '"You must define the following variables to set up MSSQL" in
+              {{ ansible_failed_result.msg.0 }}'
```

Comment:
The proper way to do this with `that: >-` or `that: |-` is
```yaml
             that: >-
                "You must define the following variables to set up MSSQL" in
                ansible_failed_result.msg.0
```
Since you are using `>-` you don't need to enclose the test in single quotes.

You don't need to use `{{ }}` here because ansible knows that the value of a `when` or `that` condition is jinja code - you can think of the value of a `when` or `that` is like a small jinja program - so it will assume the "bare" string `ansible_failed_result.msg.0` should be evaluated as a jinja variable.

This is one of those weird things about ansible - the semantics of `when` and `that` are different than other keywords.